app/services/region_backfill.rb

```diff
@@ -44,28 +44,37 @@ def create_regions
     end
     root_type = find_region_type("Root")
     org_type = find_region_type("Organization")
-    facility_group_type = find_region_type("FacilityGroup")
+    state_type = find_region_type("State")
+    district_type = find_region_type("District")
     block_type = find_region_type("Block")
     facility_type = find_region_type("Facility")
 
     root_parent = NullRegion.new(name: "__root__")
-    instance = create_region_from name: current_country_name, region_type: root_type, parent: root_parent
-
-    Organization.all.each do |org|
-      org_region = create_region_from(source: org, region_type: org_type, parent: instance)
-
-      org.facility_groups.each do |facility_group|
-        facility_group_region = create_region_from(source: facility_group, region_type: facility_group_type, parent: org_region)
-
-        facility_group.facilities.group_by(&:zone).each do |block_name, facilities|
-          if block_name.blank?
-            count_invalid(block_type)
-            logger.info msg: "skip_block", error: "block_name is blank", block_name: block_name, facilities: facilities.map(&:name)
-            next
-          end
-          block_region = create_region_from(name: block_name, region_type: block_type, parent: facility_group_region)
-          facilities.each do |facility|
-            create_region_from(source: facility, region_type: facility_type, parent: block_region)
+    root_region = create_region_from name: current_country_name, region_type: root_type, parent: root_parent
+
+    Organization.all.map do |org|
+      hierarchical_facilities = org.facilities.each_with_object({}) { |facility, result|
+        result[facility.state] ||= {}
+        result[facility.state][facility.district] ||= {}
+        result[facility.state][facility.district][facility.zone] ||= []
+        result[facility.state][facility.district][facility.zone] << facility
+      }
+
+      org_region = create_region_from(source: org, region_type: org_type, parent: root_region)
+      hierarchical_facilities.map do |state, districts|
+        state_region = create_region_from(name: state, region_type: state_type, parent: org_region)
+        districts.map do |district, zones|
+          district_region = create_region_from(name: district, region_type: district_type, parent: state_region)
+          zones.map do |zone, facilities|
```

Comment:
Can you do a `gsub(/zone/, 'block')` in this file? All the names here were `block` in here previously, which will make it easier to keep things consistent and also to find in the logs as we go forward.

Reply:
I left these ones to be zones intentionally, since they represent the Facility field that are being copied over to the region `block`. Feel free to change.